###Review Subject: Genomic Evolution of ancient Pathogens###Outline: Insights of pathogen genomic evolution from ancient DNA###Authors: Arriola LA, Weyrich LS, and Cooper A###Journal: PloS Pathogens#Abstract##Background 
#### IntroductionThe advent of new molecular biology technologies has brought a more profound understanding of the nature of microorganisms and their interactions with the environment and other organisms, including humans. One of the most significant advances came with the development of Next Generation Sequencing (NGS) technologies, which along with other molecular biology techniques have allowed the study of a wide variety of microorganisms that cannot be cultivated (estimated to be 99% of the total microbial diversity). #### Ancient microorganisms and palaeomicrobiology 
Ancient microorganisms are an example of uncultivable organisms to which genomic information we have gained access to thanks to NGS technologies. Palaeomicrobiology is a discipline involved in the study of microbial organisms from ancient sources. By studying DNA from ancient microorganisms (aDNA) we have been able to examine and compare their genomic content and structure, and to dig into their evolutionary history at particular stages of the past. To date, great efforts have been invested on studying a handful microbial pathogens involved in notable pandemic events, however other pathogenic microorganisms with reduced lethality have been oversighted and few analyses on commensal microorganisms with an important role in human health have not been yet examined. Also, analyses on the impact that microbial organisms (commensal and pathogenic) could have in the modeling of our immune systems have not been explored.
#### Presentation, aim and conclusions
Here we review some of the most notable cases on ancient pathogenic microorganisms studies, and explore the application of tools and analysis to other ancient microbial organisms.

	####Other points	o	First associations with pathogenesis.
	o	Previous methods of study of bacteria		â€¢	New technologies, new methods, new approaches	o	NGS improvements of the field	o	Ancient DNA and paleomicrobiology	o	Studies on ancient microbes	o	Conclusions#Introduction##Background:###*Knowledge of microorganisms is recent*####o	First associations with pathogenesis.
	In the past few years we have realised the great impact that microbial organisms have in human health. They are not only disease agents, but also provide us with substrates and enzymes indispensable for the correct development and functioning of the body.

The known existence of microorganism was first recorded more than 400 years ago (REF, Leeuwenhoek), but the association of bacteria as a source of pathogenicity is a relatively recent event in the Human History, established only at the end of the 19th century (REF,Koch). During this period, we have studied bacteria using a variety of physical and chemical methods, including microscopy and phenotypic tests(culture media, colony mprphology, and stainning methods)(REF). The recent advancements in molecular biology and the development of new DNA sequencing technologies have greatly extended our understanding of these pathogens.
Today, more than 45,000 complete -or near complete- genome sequences from a variety of bacterial organisms have been obtained(Genomes Online Database (GOLD v.5)).

Due to their involvement in the development of disease, pathogenic microbes are more widely studied and characterized at the genomic level. Extensive genomic data from these pathogenes has provided new insights into the consequences of genomic evolution and specific phenotypes. However, how bacterial genomes have evolved over expansive time scales, i.e. over the course of modern human evolution, and the phenotypic consequences of those genomic alterations on infectivity, pathogenicity, or lethality have largely remained unknown. Recently, high-throughput sequencing technologies coupled with hybridization enrichment techniques have allowed researchers to retrieve reliable genomic information of microbioal pathogens from ancient sources (REF). The burst of ancient bacterial genome sequencing, in addition to the use of metabarcoding of bacterial species in ancient samples, has resulted in the development of a new field, palaeomicrobiology - a discipline devoted to the study of ancient microorganisms(REF1). By comparing ancient and modern microorganisms, researchers are now able to reconstruct the evolutionary history of pathogens over expansive time scales and trace specific genomic changes that influenced past epidemics.

Although the application of aDNA techniques in microbiam evolution is in its infancy, this approach has already been proven to be a vital tool in microbial evolutionaty studies. In this review, we examine how the field of paleomicrobiology was developed and review the findings from ancient genomes of several key human bacterial pathogens, including *Yersina pestis*, *Vibrio cholerae*, *Mycobacterium tuberculosis*, and *Mycobacterium leprae*, to highlight how these studies have revolutionized our currents understanding of microbial evolution. We will also explore the current pitfalls and barriers within this field of study, and describe how the field of aDNA can be applied to address several epidemiological and *evolutional??* questions that remain unanswered.

###(Current Methodology and Technology in )Ancient DNA SequencingAncient DNA (aDNA) is the residual genetic record that can be found in historical and archaelogical samples. It can be tretrieved from various archaeological sources, including skin, teeth, soil, museum specimens, coprolites, dental calculus, and bones. By obtaining and sequencing aDNA, researchers can test evolutionary hyposthesis and contrast models with *real time* information(REF3)on animal and human evolution, ancestry(REF4,5), as well as past pathogenic agents and microorganisms(REF6,7). In general, there are certain inherent complications of working with bacterial aDNA, as degradation of the genetic material (which limits the amount and condition of molecules of interest), the limited amount of molecules from the organims vs. the host, and the contamination of the samples (either by modern or ancient DNA from other sources)(REF1) create certain additional hardships. Therefore, validation of the results on aDNA is critical and requires the implementation of certain basic standards, such as the use of separate work areas, absence of amplicons on negative PCR controls, proper molecular behaviour, originality and consistency of the amplicons, and replicability of the identification from sam and different extracts of the sample in both the same lab and in independent laboratories(REF8-10).

Ancient DNA research had a turbulent start due to these limitations, but the advent of new DNA sequencing technologies and the application of more stringent standards has allowed researchers to trace the evolutionary histories of humans, megafauna, and micrommamals, as well as microorganisms, providing, critical insight into how organisms adapt and evolve over time(REF11-14).

###Palaeomicrobiology
As in other fields of microbiology, the initial palaeomicrobiology studies relied on microscopy and the use of basic biomolecular techniques and assays (***LIST***). Due to DNA damage, ancient microbes cannot typically be cultured or grown in laboratory settings, limiting the ability to conduct these assays on ancient specimens. As one of the largest revolutins of molecular biology, the invention of the polymerase chain reaction drastically altered the ability of researchers to identify and examine ancient microbes. This allowed researchers to minimally explore the genetics of ancient microbes by screening for specific DNA sequences without the need to culture. However, this technique is highly sensitive to contamination (REF), which led to erroneous conclusions early on and brewed mistrust in the field, highlighting the necessity fro increased validation and more sensitive methods to be developed. In (**YEAR**), paleomicrobiology was revolutionized by the implementation of high-throughput sequencing, such as Illumina. THe short read lenghts provided by Illumina mathces well with the degraded nature of short ancient DNA fragments. This improved DNA sequencing technology allowed researchers to examine the diversity and complexity of bacterial populations for the first time (REF2,12,15-18).

Once high-throughput DNA sequencing methods were developed, several studies examining microbial community diversity were conducted. Whole genome shotgun sequencing (WGS) was applied in ancient permafrost samples for the first time in (***TIME***). Since then, additional metagenomic sapproaches, such as metabarcoding, have also been applied to explore the diversity of bacteria present in ancient skeletons (REF21-22). Using metabarcogding of calcified dental plaque (calculus), i.e. a preserved oral bacteria biofilm present in ancient teeth, Adler *et al. were able to describe major alterations in the composition of oral microbial communities through time (REF21), which were associated with several historical, cultural, and dietary revolutions. Using this approach, the outhors also observed changes in the presence of two specific orgal pathogens through time, *Streptococcus mutans* and *Porphyromonas gingivalis*. This research has since been extended to now include the reconstruction of a periodontal pathogen *Tannerella forsythia* (REF22) from WGS data. Despite the typical small size and number of archaeological samples, these studies revealed the great potential of paleomicrobiological studies to examine the evolution of microbial organisms, given its extraordinary preservation of biomolecular information (REF).

As a consequence of the sharp decrease of sequencing costs owed to NGS techniqeues, hybridization capture or enrichment methods were also designed and employed to examine ancient microbes. THis approach involved creating RNA and DNA 'bait' sequences that are used to 'fish out' sequences of interest. These methods markedly extending the ability to obtain genomic information from a single species present within a complex 'soup of bacterial and environmental DNA (REF19). To date, researchers have reconstructed draft genome sequences from several key ancient microorganisms, including *Vibrio cholearae*, *Tannerella forsythia* (REF22), *Yersinia pestis*, *Mycobacterium tuberculosis*, and *Mycobacterium leprae* (REF23,24). Whole genome comparisons of ancient microbial pathogens con be used to infer complex population dynamics, investigate environmental and host-pathogen interactions, and test hypotheses about the evolution of virulence through episodes of selection and extinction (REF26-29). In the following section, we examine and summarize some of the most relevant research on these ancient pathogens assesing approaches that have been used for the study of their genomic evolution and examine the main research targets posed by palaeomicrobiology.

###Previous Palaeomicrobiology Studies
####*Yersinia pestis*: Identifying the infectious agent of Black Death and Plague
One of the earliest palaeomicrobiology studies was to identify the cause of the "Black Death" (BD), whcih killed millions of people in the 14th century. Many researchers proposed *Yersinia pestis* as the cause of the disease based on symptom descriptions, although this had not been verified using molecular techniques. The first attempts to determine the pathogen responsible of these epidemics using aDNA were performed in 1997 by Michel Drancourt, a pioneer in palaeomicrobiology. Using primers specific for *Y. pesist* his team amplified and sequenced aDNA from teeth of BD(REF). HOwever, years later, a team from the Centre for Geo-Genetics in Denmark attempted the isolation of the pathogen from burials of victims of the plague following the guidelines established for aDNA studies, and they were not successful in retrieving aDNA from *Y. pestis*. To settle the debate, Hendrik Poinar and Johannes Krause (REF29,30) led novel attempts to obtain *Y. pestis* from plague specimens. These results were performed in two separate labs and follwoed strict validation guidelines to demonstrate aDNA authenticity. Using sequences from modern *Y. pestis* to perform hybridization capture and enrichment of analogous aDNA sequences, bothe teams were able to retrieve and reconstruct the ancient genomic sequence of *Y. pestis* from plague victims, finally convincing the scientific community that this bacterium was the infectious agent of the BD. Further research conducted by Krause's research team used phylogenetic comparisons to modern strains to examine the origin of the ancient strain. this analyis identified the origin of the BD plague from a previous pandemic event, the Justinian Plague from the 6th century(REF). Recently, whole genome sequence of the Justinian strain has been published (REF31). allowing the comparison of three ancient genome sequences involved in three different pandemic events. This information has been used to trace the distribution and dynamics of this pathogen through time and around the world. These studies of *Y. pestis* also demonstrated very little variability between ancient and modern *Y. pestis* strains, suggesting that the pathoegenicity associated with this disease could be due to other environmental factors, although these remain unknown.

These studies were critical in developing the laboratory methodologies of hybridization enrichment for palaeomicrobiological implementation, demonstrating how effective hybridization capture and enrichment techniques can be applied to ancient microbial DNA. For the first time, researchers were able to trace bacterial evolution temporally and geographically in ancient civilizations. However, the outcomes of these studies are much wider reaching. Ancient DNA analysis of *Y. pestis* confirmed of the infectiouse agent of the Black Deathc, identified the possible origin of the ancestral strain of this pathogen before the disease moved into humans (REF29), and identified *Y. pestis* as the cause of the Justinian Plague and the Europe plague in the 14th century (REF32) as two independent emergence events (REF31). These studies represented the first successful approach to sequencing ancient draft microbial genomes, and revealed that we knew very little about how pathogens evolve alongside humans in real-time.

####*Mycobacterium tuberculosis*: Origins and migrations of highly conserved pathogenic bacterial genomes
Due to their unique characteristics, suchs as thicker-than-average cell wall and high G-C content, *Mycobacterium* species have the most highly conserved ancient genome of any bacterial species (REF33,34). *Mycobacterium tuberculosis*, one of the most prevalent pathogens in human history (REF35), is the second largest cause of infectiuos disease death (REF 36-38). Along with other genetically closer *Mycobacteium* species, collectivelly know as *Mycobacterium tuberculosis* comples (MTBC), it is the predominant aetiological agent in the development of tuberculosis (TB). *M. tuberculosis* is the most widely studied pathogen via aDNA in palaomicrobiology (REF28), having been identified in hundreds of individuals from various geographical regions and across several times in history.
*M. tuberculosis* was the first ancient bacterial pathogen to be studied (REF39) by molecular methods, and initial palaeomicrobiological research on *M. tuberculosis* concentrated efforts on the identification of the pahtogen via PCR in ancient individuals that displayed osteological damage indicative of tuberculosis (REF39,40). Early attempts focused on identifying ribosomal proteins and insertional sequences specific to MTBC (fixREFs?). Further characterization was performed by spoligotyping (spacer oligotyping) techniques and restriction fragment polymorphism (RFLP) analysis -two approaches that were later criticized for their unsuitability for phylogenetic analysis, their poor reproducibility, and limitations in aDNA studies(REF41). More recent and accepted apporaches include the use og PCR to target single nucleotide polymorhphisms (SNPs) (REF35,38) and large sequence polymorphisms (LSP)(REF35).
PCR evidence of aDNA from *M. tuberculosis* obtained from several thousands-year-old samples (REF33,42-44) suggests a long-term co-ocurrence of the pathogen in human populations. Most of these studies characterised a few loci within MTBC and provided evidence of the emergence of certain TB strains though time; however, they failed to provide further information about specific genomic changes that differentiate such strains and traits that could have been involved in their pathogenicity(REF45). Further, the loci utilized for characterization have since been shown to be conserved through both the human infecting lineages of Mtb and soil isolates, making identification from buried ancient samples spurious (REF). In light of this, hybridization enrichment techniques have since been utilized to identify the origins and examine its recent history. For example, phylogenetic analysis refuted the once accepted hypothesis that suggested MTBC originated during a zoonoses event from *M. bovis* the causative agent of TB in cattle (REF46-48).
Recent studies have expanded the analysis of ancient *M. tuberculosis* to examine human migration and demographic changes by analysing few sequences from a variety of geographical areas and across different times (REF24,49). 	Nevertheess, a greater number of loci or even whole genome sequence across several individuals should be targeted to asses an adequate resolution that allows to discern complex relationships or migratin histories. Some analyses have achieved this by incorporating NGS analysis and enrichment capture methods to obtain whole genomi sequences. These methods have already offered a possible explanation to a question that has been pussling researchers for a long time regarding the origin of ancient tuberculosis in the Americas.

In a recent study, whole genome sequences were used to compare modern and ancient *M. tuberculosis* genomes (REF50,51) in South America, Europe, and elsewhere, shedding light into the possible oriin of TB in pre-Columbian South Aerican individuals. COmparisons of South American human TB isolates were most closely related to TB isolates from seals, suggesting a zoonotic transmissioneent in South American humans from a *M. tuberculosis* strain specific of seals, followed by a later re-adaptation and dispersal in humans(REF50).

Research on ancient M. tuberculosis **IS ON THE RIGHT TRACK?** including whole genome sequences to elucidate the evolutionary history, orignis and migrations of modern lineages. However, the integration of additional whole genome sequences (ancient and modern) from *M. tuberculosis*, from different geographical sources and convering a large time-transect, could be used to increase our resolution in time to be able to appreciate small events and address questions that are elusive to our current models, improving our understanding of the evolution and distribution in human populations through the world and across time.

####*Mycobacterium leprae*: MAssive gene decay, low genomic variation, an unexplicable decrease on prevalence
*Mycobacterium leprae* is the infectious agent of leprosy, and alongside TB, has likely infected humans since prehistoric times (REF52). Interestingly, archaeological evidence, such as the establishment of several leper colonies all over Europe, indicates leprosy was a scourge in the Middle Ages which inexplicably started its decline circa the 18th century(REF52). *M. leprae* is an obligate intracelullar pathogen and has slow growth, high genomic conservation between strains, a great number of pseudogenes, and lost several housekeeping genes (REF53). These features suggests that the organisms has undergone genome reduction or degradation similar to the observed in intracellular symbiosis(REF54,55).
Palaeomicrobiological studies have identified genes  specific to *M. leprae* (in contrast to *M. tuberculosis*) in bone samples; however, the great majorty of these studies did not comply with the criteria required for their authentication, rendering their results inadmissible. Recently complete genomes of ancient *M. leprae* have been recovered from extractes from European skeletal remains that exhibited lesions charactersitic of leprosy (REF 52,55). In addition to following the basic guidelines for authentication of aDNA, ancient human mtDNA was laso enriched along the samples as an additional control for their authenticity. The recovered mtDNA had distinctive damage patterns consistent with aDNA, but the aDNA from *M. leprae* showed a lower-than-expected damage, highlighting questions regarding the ancient nature of the entire enriched bacterial genome or the decay kinetics of thick outer walled *Mycobacterium* species. Nevertheless, this study was able to obtain two draft *M. leprae* genomes and found very few sites of differentiation in comparison to modern strains. The major finding was the observation of drastic genome reduction and a large number of pseudogenes thought to play a role in *Mycobacterium virulence*. These charnges were speculated to be a consequence of a recen bottleneck. Therefore, *M. leprae* represents an interesting study case to observe evolution and lineage extinction. Phylogenetic analyses on these two ancient strains found them to group whitin modern Near East and Asian strains, as well as a proposed ancestor of the most recurrent strain in ancient and moder specimens of Western Europe. This suggests that these strains, and likely all of European leprosy, originated in Asia(REF55). These researchers also found little to no genetic evidence for virulence reduction in modern *M. leprae*, so hypotheses derived from differences in ancient or modern leprosys outbreaks were likely influenced by alternative factors, such  as altered living conditions, changes in human immunity, or even other epidemics or co-infections, likely influenced reduced incidence of leper cases in Europe. NA alternative explanation could be the coexistence of *M. leprae* and *M. tuberculosis* and that this interaction was key in the reductionof cases of leprosy, due to a possible higher mortality.

Modern and ancient DNA comparisons showed a low level of genomic variation and suggested a recent (~5,000 years ago) acquisition in the human lineage. Interestinglu, a close relative, *M. lepromatosis*, with similar clinical profile, appeared to have been diverged from *M. leprae*  several millions ago. Further work should explore and compare additional whole genomic sequences of various strains of both microorganism through time to explore their evolutionary history(REF57). If specific genes associated with documented epidemics could be identified, these could be later used as targets for treatment of this disease.

####*Vibrio cholerae*: Identification of strains involved in ancient pandemic events through whole genome sequence

Cholera is an instestinal disease caused by infection with *Vibrio cholerae*. In the past 200 years, seven cholera pandemics have been recorded worldwide, which were attributed to two biotypes(serotype O1: classical and El Tor(REF58)). Pandemics that have occurred in the last 100 years have been genetically characterised, but alternative investigations based on disease aeteology of earlier epidemics suggest that the O1 classical biotype may have been dominant before the last century. To test this hipothesis, Devault presented a complete sequence reconstruction of the genome of an ancient *V. cholerae* sampled from a preserved sample of intestine of a patient who died in the second cholera pandemic of 1849(REF59). THis study was able to identiy the ancient strain as a "classical" biotype, confirming previous speculations. Whole genome sequence comparisons between modern and ancient strains in this study also revealed high levels  of conservation, detected several genomic rearrangements, and identified the acquisition of genomic islands in modern genomes.

It remains unclear how significant of a disease cholera was in more ancient periods. Evidence suggests that the origin of the pathogenic strains involved in the las pandemics came from Asia, so a more detailed exloration of ancient skeletons  from that area could provide important insights on its origins. Sequencing ancient DNA from other sources, such as coprolites, could help identify *V. cholerae* in other ancient specimens on populations with poor medical records. Future studies on ancient *V. cholerae* should aim to analyse dated samples from other cholera pandemic events to investigate if ancient populations were affected by similar strains of *V/ cholera*, and to date other genomic changes that could have helped in the origination of its life as a human pathogen. Moreover, more dated ancient genomes could help to establish evolutionary rates of *V. cholerae* could be by adding a temporal dimension to the studies, and to identify other genes involved in pathogenicity.

####Lessons from palaeomicrobiology studies
The study of ancinet microbial pathogens has been mainlu driven by the progress and accessibility of DNA sequencing technologies. As aDNA methodologies have improved, several palaeomicrobiology studies have identified and added a temporal dimension to the evolutionary history of important human microbial pathogens. Previous attempts to observe evolution , using only a single or a limitied number of loci to trace their phylogenetic histories, provided only limited descriptions of temporal and geographic distributions of pathogens within human populations. Thes methods can also incorrectly identify species, and did not provide insight into genome-wide changes, through time, which allow organisms to become more or less pathogenic, adapt to niches, and locate where specific strains originate. From this, it is necessary to understand the limitations of methods, despite our desire to understand evolutionary mechanisms and the past historu of bacterial pathogens.

Then invention of several next generation sequencing technologies positively affected the field of palaeomicrobiolgy, allowing researchers to amplify, target, and retrieve specific aDNA sequences from a variety of microorganisms and reconstruct nearly complete genomes. By comparing whole genomes sequences of ancient and modern microbial pathogens, researchers have now been able to describe differences between strains and interrogate their virulence mechanisms. By identifying the genomic changes in pathogenic microorganisms through time, and comparing their effectson ancient and modern populations, we become able to understand the underlying causes of pandemics and able to prevent future outbreaks.

####The future of palaeomicrobiological studies
Because of their relevance to human health, the majority of paleomicrobiological studies have concentrated on the simple diagnosis and identification of ancient pathogenic organisms. However, human interaction with bacterial organisms can also be of commensal or mutualistic nature. Similary, commensal bacteria can become pathogenic depending on the conditions, albeit those conditions are poorly understood. Therefore, it is important to examine not only pathogenes, but also the symbotic microorganisms in the body and the ecological context, i.e. chemical, nutritional, and physical environment, in which pathogens are successful. Infectivity and disease success can be highly influenced by competition (for similar sources or spaces) and cooperation, which have not yet been fully examined in paleomicrobiological studies. Until now, only a few paleomicrobiological studies ahve investigated the effect that other microorganisms have on the development of a disease and on its success(REF:TB and Leprosy) or tried to identify the microbial community in which this organisms are found(REF:Adler, Warinner). Using metagenomic, proteomic, or metabolomic approaches on ancient samples will make it possible to explore and unravel specifiec relationships and interactions that could help explain pathogen success or failure and unravel specific relationships and interactions that could help explain pathogen success or failure as in the case of the relationship between *M. tuberculosis* and *M. leprae* and identify other microorganisms that support or inhibit pathogenic microogranism, bothe in the past and today.

####Microbiomes and commensal bacteriaThe microbiome is a group of collective microorganisms found in a particular environment, and it has been recently shown to play significant roles in human health and diseas(REF). Studies on the human microbiome have shown that its composition is largely stable and resilient across the host life (REF), with several exceptions related to age and diseases. However, the composition of the human microbiome canchange rapidly in response to alterations in environment i.e. diet, disease, location, etc. (REF). Studies of modern microbiomes from individuals that live traditional hunter-gatherer or "ancestral lifestyles" have shown significantly different bacterial community composition compared to individuals in industrialized countries (REF; Italians, yanomamÃ¶). Studies on ancient microbiomes collected from calcified dental plaque (calculus) have identified similar trends at larger time scales, highlighting significant changes in bacterial composition as humans underwent large cultural changes, such as the adoption of farming or the industrial revolution (REF). Examinationsof rpeserved human faeces (Coprolites) or in very special cases, museums/hospital samples that have been properly conserved, have provided reliable information about the composition of microbiomes in samples that are only a few centuries old. Despite this, the mechanisms that determine how these changes in microbiome composition impact pathogen load and infectivity remain unexplored.
Mutualistic relationships extend beyond bacterial species competition and cooperation, as symbiotic relationships can be easily formed between the host itself and the microorganisms. For example, microorganisms can provide essential metabolic products to hteir hosts(vitamin K)(REF). These relationships are evidence of long standing relationships. Therefore, the relationships between alterations in the human genome and the micorbiome could provide critical information to ascertain the influences that microorganisms had on the human genome, or vice versa (REF:SmuAndr2015). For this purpose, samples spaning large time transects, exploring both the genome and microbiomes from ancient hominins and a multitude of ancient, closely related primates, should be completed to ascertain if and how differences in the genome correlated with the prevalence of certain microorganisms. As more ancient human genomes are sequenced/investigated.
While studying microbiomes, researchers have noticed a fair amount of variability in the content of bacterial specied. By identifien those groups of speices (by Operational Taxonomic Units, OTUs) that remains constant among helathy individuals a core microbiome has been identified. THe study of this group of orgnisms and its changes through time is of particular interest as it is essential for the identification and development of probiotics. Within these groups it is important to study the evolution of commensal bacteria that have been constant components of microbiomes sin ce ancient times and the impact that ancient disease had on them.
####Other microorganism
Paleomicrobiology studies also include research on microbial orgnaisms other thatn bacteria. Ancient yeasts, viruses, fungi and parasites provide important insights on organism that have beeen interacting with humans for a long time. In specific, the study of ancient yeast represents an importan field of research, as they have played critical roles in the fermentation and preservation of food prodcuts. Importan efforts to analyse and compile genomic information from a variety of strains from *Scaccharomyces cerevisiae*, the budding yeast have resulted in the establishment of databases sucha as the *Saccharomyces* Genome Database (SGD)(REF: Engel, 2013) associating genomes with detailed functional and phenotypic information. Similar efforts should be applied by palaeimicrobiology studies to understand the impact that human selection and interaction has had on the modern day diversity of yeas. An importan advantage of yeasts in palaeomicrobiologiacl studies is their ability to form spores, which are resistant to a variety of environmental stresses, allowing them to be detected on coprolites (REF:Cano, 2014) and other ancient sources. THe interaction of human yeast and their products, such as fermentation could have had an identifiable print on human evolution. In ancient times, with restricted sources of clean water, fermented beverages provdied an improtan source of hydration as the increase of levels of alcohol in the environment represent a natural way for yeast to eliminate competitors. Humans with the ability to process and rapidly eliminate alcohol from their bodies could have been positive selected (REF.)However we have only know about their impo	Previous methods of study of bacteriaâ€¢	Methods and technologieso	Modern bacterial genome sequencingâ€¢	Information obtained through itâ€¢	Information that cannot be obtained through ito	NGS improvements of the fieldo	Ancient DNA and paleomicrobiologyâ€¢	Ancient genome sequencingâ€¢	Information that has been obtained through itâ€¢	Information that can be obtained through itâ€¢	Current Methodology and Technology in Ancient DNA Sequencingo	Advances in technology and methods allowed to work out evolutionary relationships on microbial organismso	Ancient DNA records the genetic variability of organisms in a temporal and geographical way. â€¢	Hurdles to access ancient genomic informationo	NGS and other methodologies have allowed us to access the recorded information, allowing to explore Whole Genome information from relevant organisms.o	Studies on pathogenic microorganisms